pkg/pulumiyaml/run.go

```diff
@@ -60,6 +60,7 @@ func LoadFromCompiler(compiler string, workingDirectory string) (*ast.TemplateDe
 	cmd.Dir = workingDirectory
 	cmd.Stdout = &stdout
 	cmd.Stderr = &stderr
+	cmd.Env = append(os.Environ(), env...)
```

Comment:
According to the docs already present env-vars will be overridden:
https://pkg.go.dev/os/exec#Cmd.Env

So we should be fine

Reply:
Ah thanks, I missed that bit in the docs.

The thing I'm still curious about is which should take precedence?  The `PULUMI_*` variables from the users environment or the ones we set internally?  Right now it's the ones set internally, but I wonder if the users environment should take precedence?  

I think I could justify saying "it doesn't matter because you shouldn't be setting environment variables starting with `PULUMI_*` and expect them to do something we never specified anyway", in which case this seems fine.

Reply:
I think the scenario I'm most worried about is if a user is already using a variable with the `PULUMI_` prefix in their program currently, and this would break them.  Thinking some more about it I think I might be more worried about that then the user being confused if they do have something set in their environment and try to use this feature.

Either way I think it would be good to warn the user in this case, you can see an example of how to do that below (https://github.com/pulumi/pulumi-yaml/pull/578/files#diff-228d45f9d585e3feb8e035e1d905d4321203f645dd1a8be21da364c5699ad671R69)

Reply:
Fair enough 👍